src/main/java/org/folio/rest/impl/ItemStorageDereferencedAPI.java

```diff
@@ -0,0 +1,198 @@
+
+      asyncResultHandler.handle(Future.succeededFuture(
+        GetItemStorageDereferencedItemsResponse.respond200WithApplicationJson(itemCollection)));
+    });
+  }
+
+  @Validate
+  @Override
+  public void getItemStorageDereferencedItemsByItemId(
+    String itemId, String lang, java.util.Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
+    logger.info("id number: " + itemId);
+    if (!UuidUtil.isUuid(itemId)) {
+      respondWith400Error("Invalid UUID", asyncResultHandler);
+    }
+    String whereClause = "WHERE item.id='" + itemId + "'";
+    PostgresClient postgresClient = PgUtil.postgresClient(vertxContext, okapiHeaders);
+    postgresClient.select(sqlQuery + whereClause, asyncResult -> {
+      handleSelectFailure(asyncResult, asyncResultHandler);
+      Row row = asyncResult.result().iterator().next();
+      DereferencedItem item = mapToDereferencedItem(row);
+      asyncResultHandler.handle(Future.succeededFuture(
+        GetItemStorageDereferencedItemsByItemIdResponse.respond200WithApplicationJson(item)));
+    });
+  }
+
+  private void handleSelectFailure(
+    AsyncResult<RowSet<Row>> asyncResult,
+    Handler<AsyncResult<Response>> asyncResultHandler) {
+    
+    if (asyncResult.failed()) {
+      String errorMessage = asyncResult.cause().getMessage();
+      respondWith500Error("Can't retrive item records: " + errorMessage, asyncResultHandler);
+    }
+    if (asyncResult.result().size() == 0) {
+      respondWith404Error("No item records found.", asyncResultHandler);
+    }
+  }
+
+  private void respondWith500Error(
+    String message, Handler<AsyncResult<Response>> asyncResultHandler) {
+
+    asyncResultHandler.handle(
+      Future.succeededFuture(
+        GetItemStorageDereferencedItemsResponse.respond500WithTextPlain(message)));
+  }
+
+  private void respondWith404Error(
+    String message, Handler<AsyncResult<Response>> asyncResultHandler) {
+
+    asyncResultHandler.handle(
+      Future.succeededFuture(
+        GetItemStorageDereferencedItemsResponse.respond404WithTextPlain(message)));
+  }
+
+  private void respondWith400Error(
+    String message, Handler<AsyncResult<Response>> asyncResultHandler) {
+
+    asyncResultHandler.handle(
+      Future.succeededFuture(
+        GetItemStorageDereferencedItemsResponse.respond400WithTextPlain(message)));
+  }
+
+  private DereferencedItem mapToDereferencedItem(Row row) {
+    
+    JsonObject itemJson = (JsonObject) row.getJson("item");
+    idFields.forEach(fieldName -> {
+      if (itemJson.containsKey(fieldName)) {
+        itemJson.remove(fieldName);
+      }
+    });   
+    List<String> fieldList = new ArrayList<String>(fields.values());
+    fieldList.forEach(fieldName -> {
+      int pos = row.getColumnIndex(fieldName.toLowerCase());
+      if (row.getJson(pos) != null) {
+        itemJson.put(fieldName, row.getJson(pos));
+      }
+    });
```

Comment:
What is the purpose of this mapping?

If the item representation in a dereferenced item is the same as a regular item, why not copy the JSON completely?